src/main/java/com/tikal/jenkins/plugins/multijob/MultiJobBuilder.java

```diff
@@ -577,11 +577,13 @@ private void addBuildEnvironmentVariables(MultiJobBuild thisBuild,
                 .toUpperCase();
         String buildNumber = Integer.toString(jobBuild.getNumber());
         String buildResult = jobBuild.getResult().toString();
+        String buildName = jobBuild.getDisplayName().toString();

```

Comment:
Probably it should be <code>getFullDisplayName()</code>, because the basic name does not display folders. Or probably you may be interested to just inject <code>x_BUILD_NAME</code> and <code>x_BUILD_FULL_NAME</code>